source/adapters/level_zero/kernel.cpp

```diff
@@ -191,9 +191,28 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunch(
   ZE2UR_CALL(zeKernelSetGroupSize, (ZeKernel, WG[0], WG[1], WG[2]));
 
   bool UseCopyEngine = false;
+
+  bool UseTmpWaitList = false;
+  if (Queue->Device->useDriverInOrderLists() && Queue->isInOrderQueue()) {
+    for (uint32_t i = 0; i < NumEventsInWaitList; i++) {
+      auto EventQueue = EventWaitList[i]->UrQueue;
+      if (Queue != EventQueue) {
+        UseTmpWaitList = true;
+        break;
+      }
+    }
+  } else {
+    UseTmpWaitList = true;
+  }
+
   _ur_ze_event_list_t TmpWaitList;
-  UR_CALL(TmpWaitList.createAndRetainUrZeEventList(
-      NumEventsInWaitList, EventWaitList, Queue, UseCopyEngine));
+  TmpWaitList.ZeEventList = nullptr;
```

Comment:
what about other dispatch calls?

Reply:
Done, moved all this logic down to createAndRetainUrZeEventList(), as it is used for all dispatch calls.